File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func rightSideView(root *TreeNode) []int {
	if root == nil {
		return []int{}
	}

    res := []int{}
	q := []*TreeNode{root}

	for len(q) > 0 {
		ql := len(q)

		for i := 0; i < ql; i++ {
			c := q[0]
			q = q[1:]
			if i == ql - 1 {
				res = append(res, c.Val)
			}

			if c.Left != nil {
				q = append(q, c.Left)
			}
			if c.Right != nil {
				q = append(q, c.Right)
			}
		}
	}
	return res
}

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: 'TreeNode', p: 'TreeNode', q: 'TreeNode') -> 'TreeNode':
      def recursive(r: 'TreeNode') -> 'TreeNode':
        if not r:
          return r
        
        if p.val < r.val and q.val < r.val:
          return recursive(r.left)
        
        if p.val > r.val and q.val > r.val:
          return recursive(r.right)
        
        return r
          
      return recursive(root)
        